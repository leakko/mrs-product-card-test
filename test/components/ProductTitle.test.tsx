import React from "react";
import renderer from "react-test-renderer";
import { ProductCard } from "../../src";
import { product1 } from "../data/products";

describe('ProductTitle', () => {
  test('should show component with customized title', () => {
    const wrapper = renderer.create(
        <ProductCard.Title title='Custom Product' className="custom-class" />
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should show component with product name', () => {
    const wrapper = renderer.create(
        <ProductCard product={product1}>
            {
                () => (
                    <ProductCard.Title />
                )
            }
        </ProductCard>
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
})

