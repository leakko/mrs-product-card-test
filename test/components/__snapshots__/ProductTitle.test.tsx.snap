// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should show component with customized title 1`] = `
<span
  className="productDescription custom-class"
>
  Custom Product
</span>
`;

exports[`ProductTitle should show component with product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Sin Imagen
  </span>
</div>
`;
